Microsoft.Azure.Cosmos/src/Telemetry/ClientTelemetryProcessor.cs

```diff
@@ -39,27 +40,38 @@ internal ClientTelemetryProcessor(CosmosHttpClient httpClient, AuthorizationToke
             ConcurrentDictionary<OperationInfo, (LongConcurrentHistogram latency, LongConcurrentHistogram requestcharget)> operationInfoSnapshot,
             ConcurrentDictionary<CacheRefreshInfo, LongConcurrentHistogram> cacheRefreshInfoSnapshot,
             IReadOnlyList<RequestInfo> requestInfoSnapshot,
-            CancellationToken cancellationToken)
+            CancellationTokenSource cancellationToken)
         {
+            CancellationTokenSource serviceCancellationToken = new CancellationTokenSource(ClientTelemetryOptions.ClientTelemetryServiceTimeOut);
```

Comment:
Should this be a Linked CancellationToken?

```suggestion
            CancellationTokenSource serviceCancellationToken = new CancellationTokenSource(ClientTelemetryOptions.ClientTelemetryServiceTimeOut);
            CancellationTokenSource serviceCancellationToken = new CancellationTokenSource.CreateLinkedTokenSource(serviceCancellationToken.Token, cancellationToken);
```

Reply:
Linking service token with processor token.